EmployeeLeaveTracking/EmployeeLeaveTracking.Services/Services/LeaveRequestService.cs

```diff
@@ -378,11 +413,42 @@ public async Task<int> UpdateLeaveRequestStatus(int id, int statusId)
                 await _dbContext.SaveChangesAsync();
             }
 
+            User employeeEmail = await _dbContext.Users.FindAsync(leaveRequest.EmployeeId);
+
+            if (employeeEmail == null)
+            {
+                throw new Exception("Employee not found.");
+            }
+
+            // using the appropriate email template based on the statusId
+            string emailTemplateKey = (statusId == 2) ? "LeaveApprovedTemplate" : "LeaveRejectedTemplate";
```

Comment:
Hard coding

Reply:
Had Already added the StatusType Table in Database with Data Seeding Concept